### DISCUSSION TITLE
Incorporated Controls

### DISCUSSION THREAD

**Discussion by Ronnie180** [DeeEmm/DIY-Dyno#7] [General]
Here is some recommendations for controls to think about for us Water Brake guys.
 
#1 - Relay to turn on a squirrel cage fan mounted in front of the engine & automatically control the fan speed as the RPM's go up during the run to simulate real world test. Also a control to set at what rpm the fan should turn on/off or not run at all. Motor would be either servo or a variable speed ac or dc motor 

#2 - Built in weather station or hook up to read temperature, humidity & baro pressure for accurate readings (BME280 for example). Another option is a way to put in weather conditions manually if wanted.

#3 - Hookup for Wire Clamp, Coil or Magnetic pickup for RPM reading. 

#4 - A way to calibrate Load cell versus calibration arm length

#5 - Automatic control of water valve when Start Test button is pushed & also a way to set the speed or length of time for a pull. ( 8-13 seconds for small engines is typical for example but also depends on what size outlet orifice is in the water brake itself plus size of water valve to, typically 1/2-1 inch for small engines as well. )

#6 - Graphs to overlay different test & save data.

#7 - AFR sensor hook up for tuning & accurate readings (LSU 4.2 Bosch Sensor for example) 

#8 - Hookup for Cylinder head temperature reading with signal amplifier built in to convert Type K ring sensor low readings to 0-5 volt signal or just the simple hookup & a signal amplifier made up to convert it & added inline with sensor if needed.

#9 - A way to set what RPM to start & stop recording data.

**Comment by DeeEmm** [comment]
Great input @Ronnie180 

I've moved this to the discussion forums.

Funny I was just thinking about  2 last night when I took a look at the code. I'm actually porting the Flow Bench code as a lot of the hard work has already been done. It already incorporates the BME280 environmental sensor and so I decided it should be retained along with the air flow code as we are already calculating air density. We can then use this to accurately convert the measured values to whatever standard is required.

1 is a hard one as you will never get a fan that will deliver enough air (unless you happen to have a wind tunnel) so 'real world' conditions are not achievable in this regards. Providing fan speed control is fairly trivial however. This is just an analog output that can be hooked up to a VFD.

3 Agreed

4 will be included in the configuration parameters along with other physical characteristics such as roller diameter and weight for inertia based dyno's. The code has to cater for all different dyno styles out of the box so ultimately the end user needs a simple way to be able to set the unit up to work with their dyno. Actual calibration of the load cell can be achieved using a test weight and another set of scales to record the test weight if a calibrated weight is not available. The expected weight can be manually entered and stored.

5 & 9 Automated control is the ultimate goal - to be able to automate the process. It is worth pondering different use cases here and ways that the modulation can be mapped. I think that this also potentially ties in with point 9 as well.

6 is already part of the base code from the flow bench project so it is just a case of applying the new data streams

7 & 8 AFR along with the provision for other sensors. Ideally it should be able to cater for a set of typical sensor integrations such as 0-5v or ADC inputs. This way they could be easily customised by the end user. The current code incorporates a 4 channel ADC (Analog to Digital Converter) This will cater for AFR / CHT / MAP / Etc with suitable signal conditioning.


I did have some of these mapped out under Projects but it seems like GitHub currently have an issue with Projects and are working to resolve it. I'll add the additional items once they've managed to figure it out.

**Reply by Ronnie180** [reply]
Lol, yeah is was a few things I thought was already integrated but at the moment when I wrote it all, they was all on my mind & I was like just write them all down before I forget lol.

1 The fan part moving enough air I agree especially for car engines & I wasn't sure if it could be controlled but I seen 2 hydraulic dyno's years ago that was set up this way which is what made me think about it & both were small engine dyno's, how it worked I'm not 100% sure but the fan was also driven by a hydraulic pump & my guess is the inlet line to the main pump for the load on the engine had a tee in it & run to the inlet side for the fan pump & when you pulled the motor down to 2000 rpm's the fan was running wide open & as you did the up pull the fan would gradually slow down as you took load off the engine. It did make a slight difference in the readings versus no fan at all & the engine didn't get as hot during the pull but then again we talking manual dyno to, where you stop at each rpm a split second to read the gauge. 

4  Yeah Inertia control I just didn't mention it as it is way simpler, quick & cheaper but agree on it has to cater to them as well especially setting for roller diameter & inertia settings. 

5 & 9 I do agree with both of them tying together for sure.  I know car guys tend to do the same second pull all the time across all builds. With my manual brake I tend to spend the pull up for bigger motors so whichever setting time or seconds that is the easiest to code would probably be good. 

6 Awesome

7 This is another one that I just brought up thinking about the small engine guys but certainly agree that is has to have at least 6-8 hookups for any & all sensor reads. 

 I didn't read through the dyno project as much as i did the flow bench one, Just what I mentioned popped up when I seen dyno to so then again it was get to typing before it all slips my mind lol.

**Comment by DeeEmm** [comment]
Looks like the data is back up now. :)

**Reply by DeeEmm** [reply]
@Ronnie180 I've added the additional suggestions and credited you for them

**Reply by Ronnie180** [reply]
Oh okay. Thanks. The big shout out & thanks goes out to you though for the coding that is way over a lot of our heads & just a piece of cake to you lol.

**Comment by Ronnie180** [comment]
Sensor reads that I think we would want & be a must for big HP dyno pulls would be Oil Pressure, Water Temperature, Oil Temperature & AFR Sensor bare minimum & the others that may be a (I'd like to read this to) would be Manifold Absolute Pressure, Throttle Position Sensor, Turbo Pressure Sensor, Fuel Pressure Sensor & maybe even a Cylinder Pressure Sensor which I checked on these awhile back & boy they are a sweet piece but certainly not cheap lol.   So maybe a 6 header hookup instead of four would cover all that any would want data for as some of the sensors could just be digital gauges mounted in a panel right beside the computer screen to watch it all.

**Reply by DeeEmm** [reply]
Typically the ADC's are four channel so maybe adding two might be the go there. The ADC1115's have pin selectable addressing so using multiple units is not an issue.

As with all testing, data is key. So being able to record it rather than just view it is basically a necessity. The ESP32 has SD card support so data size is not an issue. 

We can also break out the live data value to the GUI so that physical gauges are not required.

Ultimately some kind of datalog play back would also be good.

**Comment by Ronnie180** [comment]
You exactly right & I totally agree. I forgot about the sd card on the esp32, so 6 sounds like it should be sufficient enough as it would cover the most important & a few more that you might would actually want to collect data from just for mind sake or just testing.  

Datalog play back is probably a smart move as well to, its a good way to pick up on something you may not have seen.

**Comment by DeeEmm** [comment]
With two ADS1115's we would have 8 channels which should reasonably cover whatever most people could cook up lol. These would be in addition to the RPM, Temp, Baro, Humidity and Load Cell Data as well. 

Something I've not seen is condition monitoring applied to Dyno testing. Could hook the extra channels up to a couple of vibration sensors. Complete overkill for most folks but could provide a handy baseline reference for later comparisons. There's a potential use-case for that. 

Then there's stuff like turbo rotor speed, EGTs (probably more-so than water / oil temps), NOS and WMI flow, I'm sure there's heaps more things that people would love to log. Heck it could even log HC's and NOX if that's what the use case requires. Pretty much anything can be logged via the ADC channels with the appropriate signal conditioning.

**Comment by Ronnie180** [comment]
lol You caught me on the 2 ADC's, I read it wrong. If 4 ADC's was on it, I'm sure they would certainly be used lol but 2 being 8 hookup is surely more than most I know that have dyno's.

Any data you can collect, look at & study is a huge plus when it comes to tuning so really 4 ADC's I'm sure every single one would be used lol but 2 is also better than one.

The vibration sensor now is one thing I never thought of so that's a good point on that but like you say, probably way overkill for anybody except a few until others see or hear about it & then everybody be wanting it lol. 

It's funny that several of the other sensors you mentioned I had on my mind but thought well maybe keep it simple as possible for the diy guy but with just a few more extras for the business guys, so all them other sensors I would probably keep in mind just in case lol.

**Comment by Ronnie180** [comment]
That condition monitoring is certainly something I've never heard anybody mention especially pertaining to dyno's & I been around them for 40 years lol.

**Reply by DeeEmm** [reply]
Yes very commonly used in an industrial setting for monitoring bearing condition and the like. The important aspect is getting good baseline data so that you can later compare results. Most often the monitoring is a constant thing with threshold alarms. The idea being failure prediction in advance of the actual failure event. Ideally you want to maximise the service life of the item but do not want the risk of it running to failure. Wear is often non-linear and so fixed maintenance periods can often be wasteful. Con-Mon fills the gap.

I would imagine with typical white metal crank bearings, wear could easily be detected and would manifest itself initially as an inaudible knock - easily picked up by vibration monitoring. It is a very niche use-case of course, but could be used by anyone who campaigns an engine for multiple races between rebuilds and wants some vision on engine condition. 

I think the thing with designing hardware is that you ideally need to try and cater for all potential (realistic) use cases as it's hard to change hardware after it's been made. Adding a couple of 4 channel ADCs is cheap insurance of that as they can easily be utilised for a variety of sensors / scenarios.  Then when someone in the future wants to hook up an analog laser to measure valve bounce it's mostly just a simple settings or code change. 

Imagine - we have 8 inputs. Each input has a setting in the GUI for Title and Scaling. We hook up our device / sensor, set the title and scaling i.e. "Engine Tilt", "1 unit / 50mV " and then it will log the amount it moves in degrees and add a trace to the graph with the title "Engine Tilt".  No coding required and can easily be done by the end user by following a simple step by step guide.

Using this kind of setup it's easy to cater for whatever people want to use or log, provided that the output is a standard linear 0-5v, which in the majority of cases is standard or can be easily converted with simple signal conditioning. We can also cater for non linear signals as well if we incorporate a transfer function setting. This makes the actual inputs largely sensor agnostic. And if there is some special or crazy scenario that isn't currently covered, all we need to do is to change the code to accommodate it.

Think of any crazy thing and it will most likely be able to be monitored in this way. Float bowl vibration / inlet standoff wave length / exhaust volume / reversion / tyre temperature / etc. It's just like adding con-mon data acquisition to a dyno but all in one package.

Could even be expanded to include alarm thresholds too...

Channel name - Tyre temp
Channel scaling - 1 degree / 50mV
Channel unit - Celcius
Transfer function - 2*3^2
Display in GUI - True
Max Limit - 60
Min Limit - 0
Max Alarm action - GUI Banner / Shutdown [HARD] / Shutdown [SOFT] / None
Min Alarm action - GUI Banner / Shutdown [HARD] / Shutdown [SOFT] / None

Almost endless possibilities

**Comment by Ronnie180** [comment]
I went & read a little on the conditioning & never have thought about it nor heard about it engine building wise but what little I read & you going in depth with exactly what it is & can do, I can think of several times that this would have saved a ton of time.

One example being an engine I dyno'ed or tried to dyno, it run free, blip the throttle was fine, Start a dyno pull, pull it down to 2,000 rpm & plan to go to 8,000 rpm. halfway through the pull usually around 4800-5200 it would get tight & shut off. A minute or so to walk over to it, fire it right back up & run fine again but do the same thing. First thing that came to mind was forgot to put oil in it but nope had oil in it. I pulled the engine apart 4 times looking for anything but nothing , 3rd time I totally freshened it back up, new piston, rings, wrist pin, valve springs, 2 new valves, rocker arms, gaskets, seals, new flywheel & all & same thing.

The 4th time, it was a dual ball bearing crank motor. replaced them & walla fixed it. The bearings was brand new high dollar top of the line bearings. I cut them in half looking for possibly flat spots on the races somewhere & even measured each ball in a 100 different spots it seemed like & never found anything I thought would have been cause it & just came to the conclusion that possibly one ball or race was made out of cheaper steel or matched wrong with the bearing & under load & heat, It was expanding & locking the motor up.

This conditioning would have probably caught it in crankshaft vibration I would guess so adding that I'd probably go with that as a must have to especially with all the other stuff you can do with it as well.

**Comment by Ronnie180** [comment]
A question I just thought about. I know on my SF600, If i was flowing something I knew that was going to flow less than 100 cfm's I'd use orifice plate that was closet to that flow but over for more accurate readings. With the MAF sensor, does that pertain to them as well or not?

**Reply by DeeEmm** [reply]
Interesting question.

I would say that the answer is both Yes and No. It's all to do with system resolution.

If you use a 1000cfm plate to measure something flowing 20cfm, you are trying to measure over a very small percentage of the overall range. This reduces the resolution of the system and makes it harder to determine small changes. I.e. for a 50% flow improvement the manometer will only be moving ten millimeters (assuming a 1m scale). small changes become very hard to read with the human eye. So you use a 100cfm plate and for a 50% flow improvement and the manometer is now moving 100mm and so the change is a lot easier to see. This is also true if the reading is taken by pressure sensors and not a human, the measured difference is of a small magnitude and so harder to measure as accurately as it only falls over a small area of the sensors range. If the resolution of your sensor is only 100 points, when you reduce its range to just a tenth of those points the accuracy of the system is reduced as you need a 10-15% change before the sensor will read any difference

MAF sensors continually measure the MASS of the air flowing through the sensor. Simplistically it does this by heating the air, and then measuring the temperature drop after a fixed time across a fixed distance from which it then calculates velocity. This velocity is then multipled by the area of the sensor housing to generate a flow map of the sensor. So assuming that the flow is not turbulent through the sensor, the area of the sensor does not impact its ability to measure, as the flow will be uniform through the sensor. It only needs to measure a small portion of the overall MASS to accurately determine MASS-Flow

The determining factor here is how FAST the MAF reads the MASS. Remembering that flow is proportional to diameter and so flow will slow when the diameter increases. i.e. from your test piece to the MAF. Being a time based measurement this does not decrease its accuracy. In fact the opposite is true. The accuracy / resolution will only decrease when you try to measure MASS FASTER than the sensor can read, not slower. To have an effect, this would require that the flow is orders of magnitude faster than the operating frequency of the sensor, which typically would be in the kHz range. So you would need to increase the flow at least 100x before you saw any measurable impact. I actually think that this is a conservative estimate and that it would likely be 500x or more

So with an orifice based system you will reduce system resolution considerably when measuring a flow rate just 10x slower than the current orifice, however, with a MAF based system you would need to increase flow more than 100x before you saw the same reduction in resolution.

Generally the operating frequency of the sensors is many orders of magnitudes higher than the speed of the air it needs to measure. For example the sensor will measure in kHz whereas the air variation of a frequency based MAF is between 50-150Hz.  The difference in diameter however, between a 100cfm MAF and a 1000cfm MAF is exactly the same as the difference between a 100cfm and 1000cfm orifice which is nowhere near 100x larger. So using a 1000cfm MAF to measure a 100cfm object is still viable.

Of course, when you take things to extremes like this I would say that you will be trading some system resolution against maximum flow as the flow through the MAF will only be uniform in a perfect world. So what most likely ends up being affected is the repeatability of the system. Accuracy / resolution / repeatability are three system characteristics that need to be considered and are all tied together. Accuracy alone is no good without repeatability. For this reason consideration of the design of a MAF bench, especially with regards to settling distances before and after the MAF is important if you are chasing super accurate figures (I mean lab grade stuff here). But reality is that MAFs when used in automotive applications have a very high accuracy even though they are used in far less than optimal implementations (cramped engine intake systems that are not straight). So actual repeatability and accuracy impact is likely too small to be of note.

But good practice still makes sense that the optimum MAF selection will be large enough to cater for what you need to measure, rather than the biggest available. There's little point in choosing a 1000cfm MAF if you are only measuring 20cfm components. But this is one area that needs to be validated by testing.

**Comment by Ronnie180** [comment]
I thought it might be a yes & no since like you said the MAF tubes is all cramped up with sharp turns & all  but wasn't 100% sure so awesome explanation. I guess I'll validate it when up & running then & go from there.

**Reply by DeeEmm** [reply]
Yeah MAFs are interesting devices. Technically they shouldn't work, as when you do the calcs for flow to settle to laminar, you need like 4 metres of straight pipe before and after the sensor, but they do work, and well enough to accurately meter fuel in millisecond doses, so can easily handle what we need them to do.

But, if you take a look at someone like PMAS, who manufacture performance MAF sensors and take a look at their calibration rig, it's just a MAF in the middle of about 6 metres of pipe. So they are essentially increasing accuracy by ensuring that the sensor only see's laminar flow.

**Reply by Ronnie180** [reply]
I totally agree with you on that. To me, for what they do on a vehicle with all the cramped up intake tubes & turns that some have, I can't see that flow being totally straight until it hits the maf screen so it certainly should work & probably better in our case for sure. Maybe even a little more accurate to.
